From: thermatk <thermatk@thermatk.com>
Date: Fri, 22 Jun 2018 17:05:17 +0200
Subject: kill Location, fall back to system

Prevents using Google Play Location Services on Android, & instead falls back to the OS for geolocation.

Initial Commit: https://github.com/uazo/cromite/commit/628ddf9b60dc248813b54ac21d42b3a8627df527#diff-cfa03aead1a729a3897293846e1e016c7071cf12dc9e1fbd0b995c47b50566d6 (Inherited from Bromite)
Category: Privacy
License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 services/device/geolocation/BUILD.gn                  |  1 -
 .../device/geolocation/LocationProviderFactory.java   | 11 ++---------
 2 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/services/device/geolocation/BUILD.gn b/services/device/geolocation/BUILD.gn
--- a/services/device/geolocation/BUILD.gn
+++ b/services/device/geolocation/BUILD.gn
@@ -165,7 +165,6 @@ if (is_android) {
       "android/java/src/org/chromium/device/geolocation/LocationProviderAdapter.java",
       "android/java/src/org/chromium/device/geolocation/LocationProviderAndroid.java",
       "android/java/src/org/chromium/device/geolocation/LocationProviderFactory.java",
-      "android/java/src/org/chromium/device/geolocation/LocationProviderGmsCore.java",
     ]
 
     deps = [
diff --git a/services/device/geolocation/android/java/src/org/chromium/device/geolocation/LocationProviderFactory.java b/services/device/geolocation/android/java/src/org/chromium/device/geolocation/LocationProviderFactory.java
--- a/services/device/geolocation/android/java/src/org/chromium/device/geolocation/LocationProviderFactory.java
+++ b/services/device/geolocation/android/java/src/org/chromium/device/geolocation/LocationProviderFactory.java
@@ -26,19 +26,12 @@ public class LocationProviderFactory {
 
     @CalledByNative
     public static void useGmsCoreLocationProvider() {
-        sUseGmsCoreLocationProvider = true;
+        sUseGmsCoreLocationProvider = false;
     }
 
     public static LocationProvider create() {
         if (sProviderImpl != null) return sProviderImpl;
-
-        if (sUseGmsCoreLocationProvider
-                && LocationProviderGmsCore.isGooglePlayServicesAvailable(
-                        ContextUtils.getApplicationContext())) {
-            sProviderImpl = new LocationProviderGmsCore(ContextUtils.getApplicationContext());
-        } else {
-            sProviderImpl = new LocationProviderAndroid();
-        }
+        sProviderImpl = new LocationProviderAndroid();
         return sProviderImpl;
     }
 }
--
